### DISCUSSION TITLE
History tab multiple entity selection

### DISCUSSION THREAD

**Discussion by luka6000** [home-assistant/frontend#8840] [Dashboard] [answered]
Please consider allowing multiple entities to be selected on History dashboard. It's very easy to use datetime picker for history graphs or states. But it's very common situation that I want to see more than 1 entity to compare or to follow multiple values (like thermostat temperature and valve opening).

**Comment by user-id-for-q** [comment]
I would also like this option, but I would implement it by actually removing the History tab and adding history functionality to the Developer Tools "States" tab instead since I think these two areas basically show the same thing i.e. entity states, and just differ by time i.e. now or previous. 

I also use both for debugging more than anything hence I don't think History should be a top-level tab. If I want some history data to be part of the day-to-day interface then I use a history card.

By default the 'states' page would just show states and attributes now, but then also offer to show previous states like the history tab currently does. Selecting entities could be done from a mult-select list or by typing with partial matching as with current functionality, but also extending this with wildcard and regex matching.  A time span could be set using a horizontal bar that represents the span of all data that's available, with draggable 'from' and 'to' markers to set the time span that displayed data should be restricted to. Note however, that there should always be a text-entry option as well as the mouse option since that would also allow a formula to be entered such as now-minus-3-hours or sunrise-to-sunset etc.

Finally, because tuning the page requires a bit of effort, it should offer sections where entered settings i.e. time ranges and entity sets are saved and can be shown or hidden.

(On a separate but related note, I don't think entities should have a forced view of 'state' and 'attributes' since the state is effectively  just an attribute anyway. So I would just have all references take the form <entity>[.<attribute>] where if <attribute> is not specified then the result of a template is returned so the user can define their own 'state' - by default this would just return <entity>.state. This would then mean that history 'entities' could actually be specified as a set of entity attributes with the wildcard <entity>.* referencing all of them.)

Should only take 5 minutes to implement then!

I have implemented some of this by adding my own Lovelace tab called History, making it full width and adding multiple history cards down the page - each with its own set of entities and time span. With this method however, it is a bit long-winded to add a new entity and the time span is similarly awkward to get to and then only allows a span prior to now rather than any from-to span.

**Comment by karwosts** [comment, marked as answer]
This has been implemented for some time, closing.
